coil-gif/src/main/java/coil/request/Gifs.kt

```diff
@@ -35,3 +37,25 @@ fun ImageRequest.Builder.animatedTransformation(animatedTransformation: Animated
 fun Parameters.animatedTransformation(): AnimatedTransformation? {
     return value(ANIMATED_TRANSFORMATION_KEY) as AnimatedTransformation?
 }
+
+/** Set the callback to be invoked at the start of the animation if the result is an animated [Drawable]. */
+fun ImageRequest.Builder.onAnimationStart(callback: (() -> Unit)?): ImageRequest.Builder {
+    return setParameter(ANIMATION_START_CALLBACK_KEY, callback)
+}
+
+/** Get the callback to be invoked at the start of the animation if the result is an animated [Drawable]. */
+@Suppress("UNCHECKED_CAST")
```

Comment:
Nit: I would ignore this for the whole file to avoid cluttering the method signature.

Reply:
Noted